clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -347,17 +347,21 @@ void executeCommandWithRetry(
       String topicName,
       int partition,
       IngestionCommandType command,
-      Supplier<Boolean> remoteCommandSupplier,
-      Runnable localCommandRunnable) {
+      Supplier<Boolean> isolatedProcessCommandSupplier,
+      Runnable mainProcessCommandRunnable) {
+    boolean isTopicPartitionHosted = isTopicPartitionHosted(topicName, partition);
+
+    if (command == REMOVE_PARTITION && !isTopicPartitionHosted) {
+      mainProcessCommandRunnable.run();
+      isolatedProcessCommandSupplier.get();
```

Comment:
Could you double check if in forked process side, if the process will accept drop in this situation?